### PR TITLE
Add support for empty namespaces

### PR DESCRIPTION
**Pull request by cgrellard-ledger**
### 📝 Description

Some dapps needs to connect without any required namespace
Added the ability to connect without any accounts when there is no required namespace. In this case we hide the `Accounts section` in the `Session Details` page

We also now use the `buildApprovedNamespaces` util function [from WalletConnect](https://docs.walletconnect.com/2.0/web3wallet/wallet-usage#namespace-builder)

### ❓ Context

- **Linked resource(s)**: [LIVE-9063] 

See this thread for more context : https://ledger-slack-connect.slack.com/archives/C0402NGFF9U/p1692715205483189

#### The accounts session isn't displayed when there is no accounts
Screenshot 2023-08-31 at 10 48 04

#### No required chain when trying to connect from https://se-sdk-dapp.vercel.app/
Screenshot 2023-08-31 at 10 47 49

### 🚀 Expectations to reach

_Pull Requests must pass the CI and be internally validated in order to be merged._

#### Manifest to test
```
{
  "id": "ledger-wallet-connect",
  "name": "WalletConnect",
  "url": "https://wallet-connect-live-jf802e0em-ledgerhq.vercel.app/",
  "params": {},
  "homepageUrl": "https://walletconnect.com/",
  "icon": "https://forum.zeroqode.com/uploads/default/original/2X/e/e363c6521db27335d44c1134d230b8992792dde4.png",
  "platform": "all",
  "apiVersion": "^2.0.0",
  "manifestVersion": "1",
  "branch": "stable",
  "categories": ["bridge", "defi"],
  "currencies": [
    "ethereum",
    "polygon",
    "bsc",
    "optimism",
    "arbitrum",
    "avalanche_c_chain",
    "ethereum_goerli",
    "optimism_goerli"
  ],
  "content": {
    "shortDescription": {
      "en": "WalletConnect is an open source protocol for connecting decentralised applications to mobile wallets with QR code scanning or deep linking. V2 introduces new features, including the ability to connect to multiple dapps in parallel with multiple accounts. It's important to note that not all dapps currently support V2"
    },
    "description": {
      "en": "WalletConnect is an open source protocol for connecting decentralised applications to mobile wallets with QR code scanning or deep linking. V2 introduces new features, including the ability to connect to multiple dapps in parallel with multiple accounts. It's important to note that not all dapps currently support V2"
    }
  },
  "permissions": [
    "account.list",
    "account.request",
    "message.sign",
    "transaction.sign",
    "transaction.signAndBroadcast",
    "wallet.userId",
    "wallet.info"
  ],
  "domains": ["http://*", "https://*"]
}
```

[LIVE-9032]: https://ledgerhq.atlassian.net/browse/LIVE-9032?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[LIVE-9063]: https://ledgerhq.atlassian.net/browse/LIVE-9063?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/data/methods/EIP155Data.methods.ts [hunk 1]
```diff
@@ -14,5 +14,5 @@ export const EIP155_SIGNING_METHODS = {
   ETH_SIGN_TYPED_DATA_V3: "eth_signTypedData_v3",
   ETH_SIGN_TYPED_DATA_V4: "eth_signTypedData_v4",
   ETH_SEND_TRANSACTION: "eth_sendTransaction",
-  SWITCH_CHAIN: "wallet_switchEthereumChain",
+  // SWITCH_CHAIN: "wallet_switchEthereumChain",
```
Comment (KVNLS) [inline src/data/methods/EIP155Data.methods.ts:17]: 🗑️ ?

File: src/pages/detail.tsx [hunk 2]
```diff
@@ -208,52 +208,56 @@ export default function SessionDetail() {
...
-            <CustomList>
-              {Array.from(sessionAccounts).map(([chain, accounts]) => {
-                return (
-                  <Box key={chain} mb={6} flex={1}>
-                    <Box mb={6}>
-                      <Text variant="subtitle" color="neutral.c70">
-                        {getDisplayName(chain)}
-                      </Text>
-                    </Box>
+            {Array.from(sessionAccounts).length > 0 ? (
```
Comment (KVNLS) [inline src/pages/detail.tsx:212]: Could you move `getAccountsFromAddresses` function outside of the component and return directly an array from `sessionAccounts`, please?
Today the `useMemo` dependencies do not reflect the reality as accounts is not used by the function `getAccountsFromAddresses`. By using it as a second parameter you can extract the function from the component (will not be redefined for each render) and the dependencies list of `useMemo` reflects what you use inside.

**Review by KVNLS — CHANGES_REQUESTED** [review]
I have not tested the code, but I trust that it works.
Some minor comments + Could you rebase to clean up your git history?

File: src/data/methods/EIP155Data.methods.ts (hunk 1 above)
Comment (cgrellard-ledger) [reply src/data/methods/EIP155Data.methods.ts]: Comment removed

File: src/pages/detail.tsx (hunk 2 above)
Comment (cgrellard-ledger) [reply src/pages/detail.tsx]: Done ✅

**Review by KVNLS — APPROVED** [review]
Perfect, well done!

**Pull request merged**